triton-vm/src/stark.rs

```diff
@@ -332,25 +333,30 @@ impl Stark {
 
         profiler!(start "combined DEEP polynomial");
         profiler!(start "sum" ("CC"));
-        let deep_codeword_components = [
+        let deep_codeword = [
             base_and_ext_curr_row_deep_codeword,
             base_and_ext_next_row_deep_codeword,
             quotient_segments_curr_row_deep_codeword,
-        ];
-        let deep_codeword_components = Array2::from_shape_vec(
-            [short_domain.length, NUM_DEEP_CODEWORD_COMPONENTS].f(),
-            deep_codeword_components.concat(),
-        )
-        .unwrap();
-        let weighted_deep_codeword_components = &deep_codeword_components * &weights.deep;
-        let deep_codeword = weighted_deep_codeword_components.sum_axis(Axis(1));
+        ]
+        .into_par_iter()
+        .zip_eq(weights.deep.to_vec().into_par_iter())
```

Comment:
```suggestion
        .zip_eq(weights.deep.to_vec())
```
Thanks to the trait bounds of `.zip_eq()` (outlined below), the `.into_par_iter()` will be called automatically.

```rust
// rayon::iter::IndexedParallelIterator
fn zip_eq<Z>(self, zip_op: Z) -> ZipEq<Self, Z::Iter>
where
	Z: IntoParallelIterator,
	Z::Iter: IndexedParallelIterator,
```